src/ompl/geometric/planners/rrt/src/RRTstar.cpp

```diff
@@ -368,7 +368,8 @@ ompl::base::PlannerStatus ompl::geometric::RRTstar::solve(const base::PlannerTer
                 for (std::vector<std::size_t>::const_iterator i = sortedCostIndices.begin();
                      i != sortedCostIndices.begin() + nbh.size(); ++i)
                 {
-                    if (nbh[*i] == nmotion || si_->checkMotion(nbh[*i]->state, motion->state))
+                    if (nbh[*i] == nmotion || (si_->checkMotion(nbh[*i]->state, motion->state) 
+                        && ( !useKNearest_ || si_->distance(nbh[*i]->state, motion->state) < maxDistance_) ) )
```

Comment:
Why is this change needed? Since the checkMotion call is probably more expensive, wouldn't you want to swap the order of the && operands?